pkg/version/version.go

```diff
@@ -23,3 +32,92 @@ func MaybePrintVersionAndExit(printVersion bool) {
 	}
 	log.Infof("running conduit version %s", Version)
 }
+
+type versionStatusChecker struct {
+	version string
+	client  pb.ApiClient
+}
+
+func (v versionStatusChecker) SelfCheck() (checks []*healthcheckPb.CheckResult) {
+	cliVersion := v.version
+	cliIsUpToDate := &healthcheckPb.CheckResult{
+		Status:           healthcheckPb.CheckStatus_OK,
+		SubsystemName:    "conduit-version",
+		CheckDescription: "CLI is up-to-date",
+	}
+
+	latestVersion, err := getLatestVersion()
+	if err != nil {
+		cliIsUpToDate.Status = healthcheckPb.CheckStatus_ERROR
+		cliIsUpToDate.FriendlyMessageToUser = err.Error()
+		return []*healthcheckPb.CheckResult{cliIsUpToDate}
+	}
+	if cliVersion != latestVersion {
+		cliIsUpToDate.Status = healthcheckPb.CheckStatus_FAIL
+	}
+
+	controlPlaneIsUpToDate := &healthcheckPb.CheckResult{
+		Status:           healthcheckPb.CheckStatus_OK,
+		SubsystemName:    "conduit-version",
+		CheckDescription: "control plane is up-to-date",
+	}
+
+	controlPlaneVersion, err := getServerVersion(v.client)
+	if err != nil {
+		controlPlaneIsUpToDate.Status = healthcheckPb.CheckStatus_ERROR
+		controlPlaneIsUpToDate.FriendlyMessageToUser = err.Error()
+		return []*healthcheckPb.CheckResult{controlPlaneIsUpToDate}
+	}
+	if controlPlaneVersion != latestVersion {
+		controlPlaneIsUpToDate.Status = healthcheckPb.CheckStatus_FAIL
+	}
+
+	checks = append(checks, cliIsUpToDate)
+	checks = append(checks, controlPlaneIsUpToDate)
+	return
+}
+
+func getServerVersion(client pb.ApiClient) (string, error) {
+	resp, err := client.Version(context.Background(), &pb.Empty{})
+	if err != nil {
+		return "", err
+	}
+
+	return resp.GetReleaseVersion(), nil
+}
+
+type latestVersion struct {
+	Version string
+}
+
+func getLatestVersion() (string, error) {
+	resp, err := http.Get(versionCheckURL)
```

Comment:
What do you think of making `versionCheckUrl` an attribute of `versionStatusChecker` so that we can write unit tests for it?